machida3/machida.pony

```diff
@@ -834,11 +834,48 @@ primitive Machida
 
   fun py_bytes_or_unicode_to_pony_string(p: Pointer[U8] box): String =>
     recover
-      let ps = @py_bytes_or_unicode_as_char(p)
-      let ps_size = @py_bytes_or_unicode_size(p)
-      String.copy_cpointer(ps, ps_size)
+      var ps: Pointer[U8] = Pointer[U8]
+      var ps_size: ISize = 0
+      var err: I32 = 0
+      if @py_bytes_check(p) == 0 then
+        ps = @PyUnicode_AsUTF8AndSize(p, addressof ps_size)
+      else
+        err = @PyBytes_AsStringAndSize(p, addressof ps,  addressof ps_size)
+      end
+      if (err != 0) or ps.is_null() then
+        @printf[U32]("unexpected non-string value\n".cstring())
```

Comment:
Is there a python stacktrace or python datatype trace/print we can do here to make debugging easier? It seems this should be possible as from what I can tell both `PyUnicode_AsUTF8AndSize` and `PyBytes_AsStringAndSize` will raise python errors if they encounter issues.

Reply:
Good question. I'll look around the other code to see if there is something I can use here. I didn't see it on first scan but there's plenty of code.

Reply:
```suggestion
        @printf[U32]("unexpected non-string value\n".cstring())
        print_errors()
```

Reply:
@strmpnk I tried the fancy github "suggested change" thingy... in theory, you (and any assignees) should be able to choose to commit/accept/apply it.